Sources/NIOCore/CircularBuffer.swift

```diff
@@ -222,6 +222,33 @@ extension CircularBuffer: Collection, MutableCollection {
             return end.backingIndex &- start.backingIndex
         }
     }
+
+    @inlinable
+    public __consuming func _copyContents(
+        initializing buffer: UnsafeMutableBufferPointer<Element>
+    ) -> (Iterator,UnsafeMutableBufferPointer<Element>.Index) {
+        precondition(buffer.count >= self.count)
+
+        guard var ptr = buffer.baseAddress else {
+            return (self.makeIterator(), buffer.startIndex)
+        }
+
+        let indexRanges: [Range<Int>]
+        if self.tailBackingIndex >= self.headBackingIndex {
+            indexRanges = [self.headBackingIndex..<self.tailBackingIndex]
+        } else {
+            indexRanges = [ self.headBackingIndex..<self._buffer.count, 0..<self.tailBackingIndex ]
+        }
+
+        for indexRange in indexRanges {
+            for index in indexRange {
+                ptr.initialize(to: self._buffer[index]!)
+                ptr += 1
```

Comment:
This can be unchecked (`&+= 1`) as it cannot possibly overflow.

Reply:
I don't think this infix operator can be applied to a LHS of pointer:

```
Binary operator '&+=' cannot be applied to operands of type 'UnsafeMutablePointer<Element>' and '_'
```

Reply:
TIL: [`+=` calls](https://github.com/apple/swift/blob/6c829e02a80269e2e2941ae907b901cb4265972c/stdlib/public/core/Pointer.swift#L381-L384) down [to `advance(by)`](https://github.com/apple/swift/blob/6c829e02a80269e2e2941ae907b901cb4265972c/stdlib/public/core/Pointer.swift#L288-L291) which uses `Builtin.gep_Word` to increment the pointer. No idea where we could look up the implementation of `Builtin.gep_Word` but [the assembly code shows that we already do unchecked arithmetic](https://godbolt.org/z/MoM8qjo6r).

Reply:
Yeah I forgot this was a pointer: pointers don't need unchecked math.